doc/_extensions/zephyr/domain/__init__.py

```diff
@@ -51,19 +53,14 @@
 from zephyr.doxybridge import DoxygenGroupDirective
 from zephyr.gh_utils import gh_link_get_url
 
-
-import json
-
-from anytree import Node, Resolver, ChildResolverError, PreOrderIter, search
-
 __version__ = "0.2.0"
 
 ZEPHYR_BASE = Path(__file__).parents[4]
 
 sys.path.insert(0, str(ZEPHYR_BASE / "scripts/dts/python-devicetree/src"))
 sys.path.insert(0, str(Path(__file__).parents[3] / "_scripts"))
 
-from gen_boards_catalog import get_catalog
+from gen_boards_catalog import get_catalog  # noqa: E402
```

Comment:
Is the `noqa` needed? I think import are allowed after `sys.path` calls.

Reply:
yes they are, but the `ZEPHYR_BASE = Path(__file__).parents[4]` is causing that 'algorithm' to fail. But I can always move things around, of course, it will just make the the paths used in the sys.path.insert more "obscure" thoughts?

Reply:
I think its fine to ignore for now then. I'm no expert in path resolution in python either, but I would think there'd be nicer ways. Having to do `sys.path.insert` in most files feels... icky.

Reply:
> I think its fine to ignore for now then. I'm no expert in path resolution in python either, but I would think there'd be nicer ways. Having to do `sys.path.insert` in most files feels... icky.

agreed, very icky :)
I've actually made the changes so that the noqa comment is not needed anymore. If anything, it makes the ickiness even more visible so that might encourage us to look for a better solution going forward :)